csrc/executor.cpp

```diff
@@ -92,6 +92,47 @@ std::unique_ptr<PrecomputedValues>& FusionExecutor::
   return evaluator_precomputed_values_;
 }
 
+std::string FusionExecutor::getStructuredCodeFromExternalFiles(
+    const std::string& all_external_code_paths) const {
+  // Lambda to get the external code file for a specific fusion segment
+  auto getExternalCodeFile = [](const std::string& input,
+                                int64_t index) -> std::string {
+    std::stringstream ss(input);
+    std::string token;
+    int64_t count = 0;
+    while (std::getline(ss, token, ',')) {
+      if (++count == index) {
+        return token;
+      }
+    }
+    debug()
+        << "Didn't find requested external source code. Will use generated code!\n"
+        << "Number of source code files should equal the number of fusion segments.\n"
+        << "External source code filenames should be delineated with commas, e.g.: file1.cu,file2.cu.\n";
+    return ""; // Return an empty string if no match was found
+  };
+
+  // If there are provided paths and the fusion_id is valid
+  if (!all_external_code_paths.empty() && fusion_id_ >= 1) {
+    std::string single_code_path =
+        getExternalCodeFile(all_external_code_paths, fusion_id_);
+    if (!single_code_path.empty()) {
```

Comment:
Try avoiding having too many nested levels
```suggestion
    if (single_code_path.empty()) {
      return "";
    }
```